server/raft_test.go

```diff
@@ -291,3 +291,52 @@ func TestNRGSimpleElection(t *testing.T) {
 		require_Equal(t, rn.vote, vr.candidate)
 	}
 }
+
+func TestNRGUnsuccessfulVoteRequestDoesntResetElectionTimer(t *testing.T) {
```

Comment:
Would be nice to also see a "timer resets when it should" on top of this test "timer does not reset when it shouldn't". 
But I know writing these RAFT tests is a giant pain and may not even be possible.